test/parallel/test-stream-duplex-from.js

```diff
@@ -299,3 +299,15 @@ const { Blob } = require('buffer');
     assert.strictEqual(res, 'foobar');
   })).on('close', common.mustCall());
 }
+
+{
+  // Ensure piping to duplexified async iterator without calling data does not trigger the data call
+  const cb = common.mustNotCall();
+
+  Readable.from(['a', 'b', 'c', 'd']).pipe(Duplex.from(async function*(stream) {
+    for await (const chunk of stream) {
+      cb();
+      yield chunk;
+    }
+  }));
+}
```

Comment:
I don't understand this test?

Reply:
to avoid making my mistake of starting the data flow before on data event listener is attached

Reply:
The flow should start here, the callback *should* be called.

Reply:
that's right, this is why `cb = common.mustNotCall();`

Reply:
I don't understand. You are testing the opposite of what we want?

Reply:
no, we did not attach on data listener so the stream should not start...

https://github.com/nodejs/node/issues/45533#issuecomment-1321106940

Reply:
the data listener should not be relevant here. You should not have to worry about that.

Reply:
pipeline should start the stream regardless.

Reply:
that's what I thought at first, @mcollina  WDYT?